core/controllers/skill_editor.py

```diff
@@ -139,60 +137,41 @@ def get(self, skill_id):
         self.render_json(self.values)
 
 
-class SkillEditorQuestionHandler(base.BaseHandler):
-    """Manages the creation of a question and receiving of all question
-    summaries for display in skill editor page.
-    """
-    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
-
-    @acl_decorators.can_edit_skill
-    def get(self, skill_id):
-        """Handles GET requests."""
-        skill_domain.Skill.require_valid_skill_id(skill_id)
-
-        start_cursor = self.request.get('cursor')
-
-        question_summaries, skill_descriptions, next_start_cursor = (
-            question_services.get_question_summaries_and_skill_descriptions(
-                constants.NUM_QUESTIONS_PER_PAGE, [skill_id], start_cursor)
-        )
-        return_dicts = []
-        for index, summary in enumerate(question_summaries):
-            return_dicts.append({
-                'summary': summary.to_dict(),
-                'skill_description': skill_descriptions[index]
-            })
-
-        self.values.update({
-            'question_summary_dicts': return_dicts,
-            'next_start_cursor': next_start_cursor
-        })
-        self.render_json(self.values)
-
-
 class EditableSkillDataHandler(base.BaseHandler):
     """A data handler for skills which supports writing."""
 
     GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
 
-    @acl_decorators.can_edit_skill
-    def get(self, skill_id):
-        """Populates the data on the individual skill page."""
-        skill_domain.Skill.require_valid_skill_id(skill_id)
-        skill = skill_services.get_skill_by_id(skill_id, strict=False)
-        if skill is None:
-            raise self.PageNotFoundException(
-                Exception('The skill with the given id doesn\'t exist.'))
+    @acl_decorators.can_edit_skills
+    def get(self, skill_ids):
+        """Populates the data on the individual or multiple skill pages."""
+
+        try:
+            skill_ids = skill_ids.split(',')
+        except Exception:
+            raise self.PageNotFoundException
+
+        skill_dicts = []
+        for skill_id in skill_ids:
+            skill_domain.Skill.require_valid_skill_id(skill_id)
+            skill = skill_services.get_skill_by_id(
```

Comment:
I think you shouldn't fetch one id at a time. Replace this entire section with:

```
try:      
    for skill_id in skill_ids:
        skill_domain.Skill.require_valid_skill_id(skill_id)
skills = skill_services.get_multi_skills(skill_ids)
except Exception:
    raise self.PageNotFoundException(
        Exception('The skill with the given id doesn\'t exist.'))
skill_dicts = [skill.to_dict() for skill in skills]
self.values.update({
    'skills': skill_dicts
})
```

Reply:
Done.